---
layout: single
title: "S3N2Bin: Semi-supervised learning for better binning
date: 2021-03-23
---

<style>
div.caption {
    font-size: small;
    color: #333333;
    padding-bottom:1em;
    padding-left:1em;
    padding-right:1em;
    padding-top:0em;
}
</style>

_Shaojun Pand and Luis Pedro Coelho_

<div style="padding: 1em" markdown="1">

**tl;dr** S3N2Bin is a new tool for binning (inferring MAGs,
metagenome-assembled genomes), exploiting deep-semi supervised learning. It can
assemble more high-quality genomes than other tools.

</div>

## The background: what is contig binning?

The goal is to derive metagenome-assembled genomes (MAGs). The first step
(after QC) is to assemble contigs. Since from a metagenome one can rarely
obtain genomes as a single (circular) contig, the next step is to _bin_ the
contigs: _i.e._, to group the contigs that are inferred to belong to the same
genome together into the same bin.

## The intuition: what is _semi-supervised_?

For binning, one case use reference-based methods, whereby one uses existing
genomes (assumed to be high-quality) to guide the binning. This can work well
for discovering new strains of known species, but cannot discover new species.
In contrast, _reference-independent_ methods do not rely on prior knowledge
and, thus, can discover new species. Due to this flexibility,
_reference-independent_ methods are the most-widely used methods nowadays.

The most widely used contig binning methods are _reference-independent methods_
in order to provide this flexibility. In fact, some of the most exciting
MAG-driven discoveries have been unexplored clades

However, we felt that to completely disregard reference information was to
_throw the baby out with the bathwater_ and built a model which combines the
best of both worlds, in that it can (1) use reference information to improve
binning, while still (2) recovering genomes from unknown species. This is the
meaning behind _semi-supervised_.

## The results: How well does it work? (very well)

We tested on two real datasets: One from the human gut ([Wirbel et al.,
2019](https://www.nature.com/articles/s41591-019-0406-6)) and another from an
animal gut, namely dog gut ([Coelho et al.,
2018](https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-018-0450-3)).
(Tests on an environmental dataset are on-going).

The main output is the number of high-quality (as measured by
[CheckM](http://genome.cshlp.org/content/25/7/1043) and
[GUNC](https://doi.org/10.1101/2020.12.16.422776)) MAGs.

## The tool

A test version is available on Github:
[BigDataBiology/S3N2Bin](https://github.com/BigDataBiology/S3N2Bin). It may be
a bit rough around the edges (for example, not a lot of effort has gone into
providing high-quality error messages for users) and it may still change as we
develop it, but the functionality of _gets great binning results_ should be
there.

Please use [Github issues](https://github.com/BigDataBiology/S3N2Bin/issues)
for bug reports and the
[Discussions](https://github.com/BigDataBiology/S3N2Bin/discussions) for more
open-ended discussions/questions.

